src/components/course-calendar/plan.cjsx

```diff
@@ -103,190 +106,104 @@ CoursePlan = React.createClass
-    rangeLength = rangeDuration.length('days')
-    planLabelStyle =
-      width: rangeLength / durationLength * 100 + '%'
-
-    # label should float right if the plan is cut off at the beginning of the week
-    if offset < 0
-      planLabelStyle.float = 'right'
-
-    labelClass = 'continued' unless index is 0
-
-    label = <label
-      data-opens-at={plan.opensAt}
-      style={planLabelStyle}
-      ref='label'
-      className={labelClass}>{plan.title}</label>
-
-  renderOpenPlan: (planStyle, planClasses, label) ->
-    {item, courseId} = @props
-    {isPublishing} = @state
-    {plan, index} = item
-
-    if isPublishing
-      planModal = <CoursePlanPublishingDetails
-        plan={plan}
-        courseId={courseId}
-        className={planClasses}/>
-    else
-      planModal = <CoursePlanDetails
-        plan={plan}
-        courseId={courseId}
-        className={planClasses}/>
-
-    planOnly = <div style={planStyle}
-      className={planClasses}
-      onMouseEnter={@syncHover}
-      onMouseLeave={@removeHover}
-      onClick={@syncOpenPlan}
-      ref='plan'>
-      {label}
-    </div>
-
-    if index is 0
-      # only trigger modal if this is the first component representing the plan
-      planInterface = <BS.ModalTrigger modal={planModal} ref='trigger'>
-        {planOnly}
-      </BS.ModalTrigger>
-    else
-      # otherwise, if this plan continues into the next week, don't add an additional modal
-      planInterface = planOnly
-
-    planInterface
-
-  renderEditPlan: (planStyle, planClasses, label) ->
-    {item, courseId} = @props
-    {plan} = item
-
-    linkTo = camelCase("edit-#{plan.type}")
-    params = {id: plan.id, courseId}
-
-    <div
-      style={planStyle}
-      className={planClasses}
-      onMouseEnter={@syncHover}
-      onMouseLeave={@removeHover}
-      ref='plan'>
-      <Router.Link
-        to={linkTo}
-        params={params}>
-          {label}
-      </Router.Link>
-    </div>
-
-  render: ->
-    {item, courseId} = @props
-    {publishStatus, isPublishing} = @state
-    {plan, duration, rangeDuration, offset, index, weekTopOffset, order} = item
-
-    plan.isFailed = publishStatus is 'failed'
-    plan.isKilled = publishStatus is 'killed'
-
-    durationLength = duration.length('days')
-    # Adjust width based on plan duration and left position based on offset of plan from start of week
-    # CALENDAR_EVENT_DYNAMIC_WIDTH and CALENDAR_EVENT_DYNAMIC_POSITION
-    # top is calculated by using:
-    #   weekTopOffset -- the distance from the top of the calendar for plans in the same week
-    #   order -- the order the plan should be from the bottom, is an int more than 1 when a plan needs to
-    #       stack on top of other plans that overlap in duration.
-    planStyle =
-      width: durationLength * 100 / 7 + '%'
-      left: offset * 100 / 7 + '%'
-      top: (weekTopOffset + 4 - order * 3) + 'rem'
+  removeHover: ->
+    @setState(isHovered: false) if @state.isHovered
```

Comment:
You might consider just having a single `setHover` method that would accept a boolean and hand that method off the the components in `render`.  You could still use the optimization of not calling setState unless you need to.